include/envoy/ssl/private_key/private_key.h

```diff
@@ -50,11 +48,10 @@ class PrivateKeyMethodProvider {
   virtual bool checkFips() PURE;
 
   /**
-   * Get the private key methods from the provider.
-   * @return the private key methods associated with this provider and
-   * configuration.
+   * Install the BoringSSL private key method.
+   * @param ssl a SSL connection object.
    */
-  virtual BoringSslPrivateKeyMethodSharedPtr getBoringSslPrivateKeyMethod() PURE;
+  virtual void installBoringSslPrivateKeyMethod(SSL_CTX* ssl) PURE;
```

Comment:
Do you have a plan to make a use of this for OpenSSL? If no can we just exclude this for non-boringssl build?

Reply:
Right, we don’t need it for the OpenSSL mode, and I’m ok excluding it also. Will send another PR.

Reply:
I've addressed this in my latest commits.